compiler/passes/src/code_generation/mod.rs

```diff
@@ -319,28 +319,53 @@ impl Display for AleoConstructor {
 
 #[derive(Hash, Eq, PartialEq, Clone, Debug)]
 pub enum AleoExpr {
-    Bool(bool),
     Reg(AleoReg),
     Tuple(Vec<AleoExpr>),
     ArrayAccess(Box<AleoExpr>, Box<AleoExpr>),
-    U32(u32),
-    U16(u16),
-    Literal(Literal),
     MemberAccess(Box<AleoExpr>, String),
     RawName(String),
+    // Literals
+    Address(String),
+    Bool(bool),
+    Field(String),
+    Group(String),
+    Scalar(String),
+    String(String),
+    U8(u8),
+    U16(u16),
+    U32(u32),
+    U64(u64),
+    U128(u128),
+    I8(i8),
+    I16(i16),
+    I32(i32),
+    I64(i64),
+    I128(i128),
```

Comment:
Should this also account for [signature](https://github.com/ProvableHQ/snarkVM/blob/d8e736636dd11e1cf11717977802c61184da1cb2/console/program/src/data/literal/mod.rs#L72) literals?